tests/integration/test_write_stress_csv.py

```diff
@@ -0,0 +1,78 @@
+
+import pytest
+from filecmp import cmp
+from os import remove
+
+from pyrs.dataobjects.fields import ScalarFieldSample
+from pyrs.dataobjects.fields import StressField
+from pyrs.dataobjects.fields import StrainField
+from pyrs.core.summary_generator_stress import SummaryGeneratorStress
+
+
+def test_write_csv_empty_strain_filenames():
+
+    with pytest.raises(RuntimeError):
+        # strain that doesn't come from a project file
+        X = [0.000, 1.000, 2.000, 3.000, 4.000, 5.000, 6.000, 7.000, 8.000, 9.000]
+        Y = [0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000]
+        Z = [0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000]
+
+        def strain_instantiator(name, values, errors, x, y, z):
+            strain = StrainField()
+            strain._field = ScalarFieldSample(name, values, errors, x, y, z)
+            return strain
+
+        strain11 = strain_instantiator('strain',
+                                       [0.000, 0.001, 0.002, 0.003, 0.004, 0.005, 0.006, 0.007, 0.080, 0.009],
+                                       [0.000, 0.001, 0.002, 0.003, 0.004, 0.005, 0.006, 0.007, 0.008, 0.009],
+                                       X, Y, Z)
+        strain22 = strain_instantiator('strain',
+                                       [0.000, 0.001, 0.002, 0.003, 0.004, 0.005, 0.006, 0.007, 0.080, 0.009],
+                                       [0.000, 0.001, 0.002, 0.003, 0.004, 0.005, 0.006, 0.007, 0.008, 0.009],
+                                       X, Y, Z)
+        strain33 = strain_instantiator('strain',
+                                       [0.000, 0.001, 0.002, 0.003, 0.004, 0.005, 0.006, 0.007, 0.080, 0.009],
+                                       [0.000, 0.001, 0.002, 0.003, 0.004, 0.005, 0.006, 0.007, 0.008, 0.009],
+                                       X, Y, Z)
+
+        stress = StressField(strain11, strain22, strain33, 200, 0.3)
+        SummaryGeneratorStress('dummy.csv', stress)
+
+
+def test_write_csv_none_stress():
+
+    with pytest.raises(RuntimeError):
+        SummaryGeneratorStress('dummy.csv',  None)
+
+
+def test_write_csv_incorrect_filename():
```

Comment:
although not required, you could use fixture `test_data_dir` instead of hardcoding the location of the testing data (`test/data`). See, for instance:
https://github.com/neutrons/PyRS/blob/master/tests/integration/pyrs/dataobjects/test_fields.py#L190